cookbooks/child-workflows/child-workflows-megafood.md

```diff
@@ -0,0 +1,321 @@
+
+Let's look at how that is implemented in this workflow.
+
+```java
+    // ...
+
+    // Wait for an ETA or abort if restaurant rejected order
+    Workflow.await(() -> etaInMinutes != -1 || OrderStatus.REJECTED.equals(currentStatus));
+
+    // ... 
+```
+Here our workflow is **polling** until the order ETA is updated, so how is that happening? 
+
+(For more details on polling in cadence workflows, have a read of our [polling cookbook](https://github.com/instaclustr/cadence-cookbooks-instafood/blob/main/cookbooks/polling/polling-megafood.md))
+
+Lets look at our parent workflow and its **interface definition**, where it has defined the following method:
+
+**Instafood workflow - interface**
+```java
+  @SignalMethod
+  void updateEta(int estimationInMinutes);
+```
+The **@SignalMethod** annotation decorates the method and informs Cadence that this method is used to receive signals from an external process, in this case it's coming from the child workflow, which we can see here:
+
+**Megaburgers workflow - Child workflow**
+```java
+  private OrderWorkflow getParentOrderWorkflow() {
+      String parentOrderWorkflowId = Workflow.getWorkflowInfo().getParentWorkflowId();
+      return Workflow.newExternalWorkflowStub(OrderWorkflow.class, parentOrderWorkflowId);
+  }
+    
+  OrderWorkflow parentOrderWorkflow = getParentOrderWorkflow();
+
+  // Send ETA to parent workflow
+  parentOrderWorkflow.updateEta(getOrderEta(orderId));
+
+  // ...
+```
+Let's break this down a bit:
+1. First, we create a stub to our parent workflow. We do this by calling the Cadence SDK to get the ID of the parent workflow, the create the stub for it.
+2. Now that we have the stub, we gain access to the methods on it. We call **updateEta** and provide the ETA, which gets sent asynchronously.
+3. This child workflow continues as the order is being prepared.
+4. Our parent workflow receives the signal, and then it can also progress.
+
+### Reflection - Instafood and child workflows
+
+Above is a good example of how to use child workflows, and it's also a great example of **why** you want to use them.
+
+Here we are calling a child workflow asynchronously and letting it signal to the parent workflow when the important information is ready. 
+The parent workflow can keep working until this information is ready.
+This is cruicial for a workflow such as this, where we want to inform the customer of an ETA or dispatch a courier, but it doesn't make sense for the restaurant preparation workflow to have this responsibility.
+
+Similarly, the primary workflow doesn't concern itself with how each restaurant implements the various requirements for updating ETA and status. This makes the workflow simple to implement and easy to understand.
+## Running a Happy-Path Scenario
+
+To wrap-up, let’s run a whole order scenario. We are going to use the **cadence cli** to demonstrate how it can be used to start workflows. 
```

Comment:
Where does the cadence CLI get used to start the workflow?

Reply:
i was originally going to show how to use the CLI but this workflow won't progress without the api calls that the test makes, so ill revert it to the original.